[espnet-3] Merge master into espnet3 and fixed CI

Masao-Someki opened the pull request:
## What did you change?

Merge master into espnet3 branch
It contains a required patch for pyopenjtalk to fix CI.

- Removed dependency of `underthesea` from `task-tts` group.

---

## Why did you make this change?

To fix CI issue in espnet3 branch

- The `underthesea` is not yet supporting installation via pyproject.toml

---

## Is your PR small enough?

No, but no other changes from merging master

---

## Additional Context
-

sw005320 commented:
@Masao-Someki, can you review your fix for the conflict?
I think this includes some mistakes

Masao-Someki commented:
I re-checked the merges, and the only changes I made while resolving conflicts were the following.
I’m not sure why the CI errors are happening now, but it seems they’re related to a pip dependency issue...

```
diff --cc .github/workflows/ci_on_ubuntu.yml
index 45e72a7404,a8d536bb39..3588c315e7
--- a/.github/workflows/ci_on_ubuntu.yml
+++ b/.github/workflows/ci_on_ubuntu.yml
@@@ -176,9 -175,9 +176,9 @@@ jobs
        max-parallel: 20
        matrix:
          os: [ubuntu-latest]
-         python-version: ["3.10", "3.11"]
 -        python-version: ["3.10"]
 -        pytorch-version: ["2.6.0"]
 -        chainer-version: ["6.0.0"]
++        python-version: ["3.10", "3.12"]
 +        pytorch-version: [2.5.1, 2.6.0, 2.7.1, 2.8.0]
 +        chainer-version: [6.0.0]
          use-conda: [false]
      steps:
        - uses: actions/checkout@master
@@@ -194,10 -193,7 +194,10 @@@
          run: |
            sudo apt-get update -qq
            # NOTE(kamo): g++-7 doesn't exist in ubuntu-latest
--          sudo apt-get install -qq -y cmake libsndfile1-dev bc sox ffmpeg
++          sudo apt-get install -qq -y cmake libsndfile1-dev bc sox ffmpeg libjpeg-dev
 +      - name: Get PR labels
 +        id: pr-labels
 +        uses: joerick/pr-labels-action@v1.0.9
        - name: install espnet
          env:
            ESPNET_PYTHON_VERSION: ${{ matrix.python-version }}
```

Masao-Someki commented:
Okay now I understand the error. Previously we used `actions/cache@v3` which contains the wheel for `underthesea`, but since we changed to the newer version (v4) we cannot use the cache.
We can still install the package with `setup.py`, but it seems the `underthesea` does not support installation with `pyproject.toml` as we encounter dependency error with pip. (building this package with pyproject.toml installation requires `maturin` but it is not in the build requirement.)

Masao-Someki commented:
In ESPnet-2, this package was only used in a special case (Vietnamese text cleaning) and was therefore not included as a dependency. In ESPnet-3, we introduced task-specific dependencies and added this package to the `task-tts` group. Since it now causes CI issues and is only needed for Vietnamese TTS, I will remove it from `task-tts`. I will update the warnings and tutorials for users who require Vietnamese support.

Masao-Someki commented:
@sw005320 
Hi, if the above commit is okay to you, I will merge this and back to the main espnet-3 development!

sw005320 commented:
Thanks!
Now, we fix the CI error and can also use the latest numpy/python setup.
Please import the update in the main repo frequently
